src/api_server.cc

```diff
@@ -139,4 +147,35 @@ void MetadataApiServer::HandleMonitoredResource(
   }
 }
 
+void MetadataApiServer::HandleHealthz(
+    const HttpServer::request& request,
+    std::shared_ptr<HttpServer::connection> conn) {
+  std::set<std::string> unhealthy_components;
+  if (health_checker_ != nullptr) {
+    unhealthy_components = health_checker_->UnhealthyComponents();
+  }
+  if (unhealthy_components.empty()) {
+    if (config_.VerboseLogging()) {
+      LOG(INFO) << "Healthz returning 200";
+    }
+    conn->set_status(HttpServer::connection::ok);
+    conn->set_headers(std::map<std::string, std::string>({
+      {"Content-Type", "text/plain"},
+    }));
+    conn->write("healthy");
+  } else {
+    if (config_.VerboseLogging()) {
+      LOG(WARNING) << "Healthz returning 500";
+    }
+    conn->set_status(HttpServer::connection::internal_server_error);
+    conn->set_headers(std::map<std::string, std::string>({
+      {"Content-Type", "text/plain"},
+    }));
+    conn->write("unhealthy components:\n");
+    for (const auto& s : unhealthy_components) {
```

Comment:
[Optional] This is a component, so wouldn't `c` (or even `component`) work better?